packages/dd-trace/src/profiling/profiler.js

```diff
@@ -91,7 +91,10 @@ class Profiler extends EventEmitter {
         if (!profile) continue
 
         profiles[profiler.type] = await profiler.encode(profile)
-        this._logger.debug(`Collected ${profiler.type} profile: ` + JSON.stringify(profile))
+        const profileJson = JSON.stringify(profile, (key, value) => {
```

Comment:
This should be in the `log.debug()` function so that it's not executed when logging is disabled.